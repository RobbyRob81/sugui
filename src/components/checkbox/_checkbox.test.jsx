import * as React from 'react';
import { customRender, fireEvent } from '../../../test/test-utils';
import Checkbox from '.';

it('should render without throwing an error', () => {
  const { container } = customRender(<Checkbox>Test</Checkbox>);
  expect(container.firstChild).toMatchSnapshot();
});

it('calls "onChange" prop when the checked state changes', () => {
  const onChange = jest.fn();
  const { getByText } = customRender(<Checkbox onChange={onChange}>Test</Checkbox>);

  fireEvent.click(getByText(/Test/i));
  expect(onChange).toHaveBeenCalled();
});

it('should contain a text label', () => {
  const { getByText } = customRender(<Checkbox>Test</Checkbox>);
  expect(getByText(/Test/i)).toBeVisible();
});
