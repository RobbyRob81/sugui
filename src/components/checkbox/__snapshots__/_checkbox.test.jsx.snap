// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without throwing an error 1`] = `
<div
  aria-checked="false"
  class="sc-bdVaJa JwxJm"
  role="checkbox"
  tabindex="0"
>
  <div
    class="sc-bwzfXH bUjTdr"
    color="red"
  />
  <div
    class="sc-htpNat hSkRsC"
  >
    Test
  </div>
</div>
`;
